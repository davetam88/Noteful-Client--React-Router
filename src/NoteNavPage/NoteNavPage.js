import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import CircleButton from '../CircleButton/CircleButton'
import './NoteNavPage.css'
import PropTypes from 'prop-types';


export default function NoteNavPage(props) {

  return (
    < div className='NoteNavPage' >
      <CircleButton
        tag='button'
        role='link'
        onClick={() => props.history.goBack()}
        className='NoteNavPage__back-button'
      >
        <FontAwesomeIcon icon='chevron-left' />
        <br />
        Back
      </CircleButton>
      {
        props.folder && (
          <h3 className='NoteNavPage__folder-name'>
            {props.folder.name}

          </h3>

        )
      }
    </div >
  )
}

NoteNavPage.defaultProps = {
  history: {
    goBack: () => { }
  },
  folder: {},
}

NoteNavPage.propTypes = {
  goBack: PropTypes.func,
  folder: PropTypes.object,
}
