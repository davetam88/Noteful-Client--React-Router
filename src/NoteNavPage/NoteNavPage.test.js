import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteNavPage from './NoteNavPage'

describe(`NoteNavPage component`, () => {
  const props = {
    folder: {
      "name": "Important"
    }
  }

  it('renders a .NoteNavPage by default', () => {
    const wrapper = shallow(<NoteNavPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a h3 with folder name when in props', () => {
    const h3 = shallow(<NoteNavPage {...props} />)
      .find('.NoteNavPage__folder-name')
    expect(toJson(h3)).toMatchSnapshot()
  })
})
