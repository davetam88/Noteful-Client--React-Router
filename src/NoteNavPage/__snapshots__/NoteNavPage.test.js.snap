// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteNavPage component renders a .NoteNavPage by default 1`] = `
<div
  className="NoteNavPage"
>
  <NavCircleButton
    className="NoteNavPage__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavCircleButton>
  <h3
    className="NoteNavPage__folder-name"
  />
</div>
`;

exports[`NoteNavPage component renders a h3 with folder name when in props 1`] = `
<h3
  className="NoteNavPage__folder-name"
>
  Important
</h3>
`;
